config/monitor_services.go

```diff
@@ -102,14 +112,16 @@ func (c *ServicesMonitorConfig) Merge(o MonitorConfig) MonitorConfig {
 	return r2
 }
 
-// Finalize ensures there no nil pointers.
+// Finalize ensures there no nil pointers. with the _exception_ of Regexp. There
+// is a need to distinguish betweeen nil regex (unconfigured regex) and empty
+// string regex ("" regex pattern) at Validate()
```

Comment:
When would a user want to use `""` as a regular expression for service names? It seems to only match empty string service names, which Consul doesn't support.

Reply:
The empty string regular expression acts similar as a wildcard, it matches on everything. You can try it out [here](https://go.dev/play/p/MsTenXQeQnw)

Reply:
Ahh that's good to know, thank you for checking me!

Reply:
TIL! I've been doing `.*` 😂

Reply:
Same here! The actual behavior of "" isn't really what I would have guessed